packages/design-system/src/components/Dialog/Dialog.jsx

```diff
@@ -64,20 +65,24 @@ export class Dialog extends React.PureComponent {
           : '';
     }
 
-    /* Send analytics event for dialog open */
-    sendAnalyticsEvent(
-      get(this.props.analytics, eventAction),
-      get(defaultAnalytics(this.eventHeadingText), eventAction)
-    );
+    if (dialogSendsAnalytics()) {
```

Comment:
Same for this

Reply:
There is _some_ risk that between `componentDidMount` and `componentWillUnmount` that the feature flag could be turned on, and then the class properties assigned here wouldn't exist. However, app teams can just be careful not to do this. I think it would be kind of weird.